### DISCUSSION TITLE
Determining when Roborock has finished cleaning

### DISCUSSION THREAD

**Discussion by rodolfomachado** [humbertogontijo/homeassistant-roborock#477] [General]
Hi,

I was wondering the best way to determine when the roborock has finished cleaning.

To be more precise. I was trying to do an automation when it finished but it's not working properly.

I tried two diffent ways:
1) When "sensor...last cleaning" changes states.
Error: sometimes it changes during the cleaning. I dont remember when but maybe when I select some rooms.

2) When main entity state changes from cleaning to returning to dock.
Error: When I ask Roborock to do mop it leaves the dock e returns to cleaning the mop. So automation fails here.

So, how can I precisely know when a cleaning finished?

**Comment by therealmrb-w** [comment]
One way that seems to work for me, when mopping, is when the trigger changes from washing the mop to charging.

**Comment by rodolfomachado** [comment]
I`m using the main status changing to returning_home or from cleaning to charging.

Other solution was using the attribute inCleaning changing to 0 and inReturning is 1.

Not the best solution, but works

**Comment by modo1302** [comment]
hey guys, is it just me or is the robot status isn't updated instantly? sometimes it takes up to a minute, and when that happens in misses the 'returning' status.

**Reply by rodolfomachado** [reply]
Most of the time it updates quickly. But last time I used I noticed it didn't show the status returning_home. I'll pay more attention

**Reply by dnikles** [reply]
I'm noticing it update slower than it was previously. It is missing the "returning" status for me too, which it rarely if ever did before.
